pkg/controller/directvolumemigration/pvcs.go

```diff
@@ -72,7 +77,41 @@ func (t *Task) createDestinationPVCs() error {
 	return nil
 }
 
-func (t *Task) getDestinationPVCs() error {
+func (t *Task) areDestinationPVCsBound() (bool, error) {
 	// Ensure PVCs are bound and not in pending state
-	return nil
+	// Get client for destination
+	destClient, err := t.getDestinationClient()
+	if err != nil {
+		return false, err
+	}
+
+	pvcMap := t.getPVCNamespaceMap()
+	selector := labels.SelectorFromSet(t.buildDVMLabels())
+	for ns, _ := range pvcMap {
+		pvcList := corev1.PersistentVolumeClaimList{}
+		err = destClient.List(
+			context.TODO(),
+			&k8sclient.ListOptions{
+				Namespace:     ns,
+				LabelSelector: selector,
+			},
+			&pvcList)
+		if err != nil {
+			return false, err
+		}
+		for _, pvc := range pvcList.Items {
+			if pvc.Status.Phase != corev1.ClaimBound {
+				t.Owner.Status.SetCondition(migapi.Condition{
+					Type:     Running,
```

Comment:
Is this enough?

What if the PVCs are not bound for 2 hours? or 2 days?

Keeping this in running forever is not ideal. Can we raise a critical condition after say 10 mins into this condition?

Reply:
Yes, I agree. Done